### PR TITLE
feat: error handling + reorg

### LINKED ISSUE #8: murmur-core/cli: error handling + more testing

### PR DESCRIPTION
**Pull request by driemworks**
Closes #8 
- adds error handling for murmur-core/lib
- more robust unit tests + edge cases
- reorgs the crates to include the cli as a binary instead of a standalone crate
     - removes the cli directory completely and makes it usable as 'murmur' rather than 'murmur-cli
- also includes general improvements to the CLI, including configurable recipient of balance transfers

### CODE REVIEW

File: core/src/murmur.rs [hunk 1]
```diff
@@ -16,42 +16,49 @@
...
+use alloc::{collections::BTreeMap, vec, vec::Vec};
 
 #[cfg(feature = "client")]
 use crate::otp::BOTPGenerator;
 
-#[cfg(feature = "client")]
-use rand_core::OsRng;
+// #[cfg(feature = "client")]
+// use rand_core::OsRng;
+
```
Comment (juangirini) [inline core/src/murmur.rs]:
```suggestion
```

Comment (driemworks) [reply core/src/murmur.rs]: removed

File: core/src/murmur.rs [hunk 2]
```diff
@@ -357,4 +413,63 @@ mod tests {
             pos,
         ));
     }
-}
\ No newline at end of file
+
+    // #[cfg(feature = "client")]
```
Comment (juangirini) [inline core/src/murmur.rs]:
```suggestion
```

**Review by juangirini — APPROVED** [review]
LGTM

File: core/src/murmur.rs (hunk 2 above)
Comment (driemworks) [reply core/src/murmur.rs]: removed
